pyro/poutine/trace_struct.py

```diff
@@ -252,6 +252,15 @@ def compute_score_parts(self):
                     warn_if_nan(site["log_prob_sum"], "log_prob_sum at site '{}'".format(name))
                     warn_if_inf(site["log_prob_sum"], "log_prob_sum at site '{}'".format(name), allow_neginf=True)
 
+    def detach(self):
```

Comment:
nit: could you rename this method to `detach_()` to follow PyTorch naming conventions for in-place ops?

Reply:
I presume it's alright to define an in-place op that _not_ a pure side effect? 
This method also returns the detached trace.

Reply:
Yes that's fine, in PyTorch, `.detach()` creates a detached copy, `.detach_()` detaches in-place, and both return the result (which in the latter case is the same object). I just thought it would be nice if your function were named similarly `.detach_()` in case someone eventually wants to implement a `.detach() = .copy().detach_()`.

Reply:
Nevermind. I changed it so it is actually just a pure side effect (in 357f529e).